packages/site/src/components/ContentBlocks.tsx

```diff
@@ -1,35 +1,130 @@
 import { useParse, DEFAULT_RENDERERS } from 'myst-to-react';
-import type { Parent } from 'myst-spec';
+import type { GenericNode, GenericParent } from 'myst-common';
+import type { NodeRenderer } from '@myst-theme/providers';
 import { useNodeRenderers } from '@myst-theme/providers';
 import classNames from 'classnames';
+import { useCellRefRegistry } from '../pages/NotebookProvider';
+import { CellRun } from './CellRun';
 
-function Block({ id, node, className }: { id: string; node: Parent; className?: string }) {
-  const renderers = useNodeRenderers() ?? DEFAULT_RENDERERS;
-  const children = useParse(node, renderers);
+function activeCodeRendererFactory(parentId: string, BaseRenderer: NodeRenderer<any>) {
+  return function ActiveCode(node: GenericNode) {
+    const code = BaseRenderer(node);
+    return (
+      <div
+        className="relative"
+        key={node.key}
+        data-cell-id={parentId}
+        data-mdast-node-type={node.type}
+        data-mdast-node-id={node.key}
+      >
+        {/* <div className="rounded bg-blue-500 text-xs text-white px-2 py-1">
+          [CODE] block id: {parentId} | node.id: {node.id ?? 'none'} | node.key:{' '}
+          {node.key ?? 'none'}
+        </div> */}
+        {code}
+        {node.kind !== 'inline' && <CellRun id={parentId} />}
+      </div>
+    );
+  };
+}
+
+function activeOutputRendererFactory(parentId: string, BaseRenderer: NodeRenderer<any>) {
```

Comment:
So we can potentially just scrap these and modify the `code` and `output` renderers to fit our needs!?